src/main/java/uk/gov/hmcts/reform/pip/data/management/service/RetryListenerForPublications.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.pip.data.management.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.RetryListener;
+import org.springframework.retry.interceptor.MethodInvocationRetryCallback;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pip.data.management.database.AzureBlobService;
+import uk.gov.hmcts.reform.pip.data.management.helpers.ArtefactHelper;
+import uk.gov.hmcts.reform.pip.data.management.models.publication.Artefact;
+
+@Service
+public class RetryListenerForPublications implements RetryListener {
+
+    @Autowired
+    AzureBlobService azureBlobService;
+
+    @Override
+    public <T, E extends Throwable> void onError(RetryContext context, RetryCallback<T, E> callback,
+                                                 Throwable throwable) {
+        String payloadUrl = ((Artefact)(((MethodInvocationRetryCallback<?, ?>) callback)
```

Comment:
Is it possible to avoid the cast here? Wondering if moving it to the method signature will work instead

Reply:
It is not possible to move it to the method signature as it must override the exact method in the base interface. I have split it up into 2 separate lines with individual castings. Hopefully a little bit clearer